python/arcticdb/toolbox/library_tool.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Union, List
+
+import pandas as pd
+
+from arcticdb.version_store._normalization import FrameData
+from arcticdb_ext.codec import Segment, decode_segment
+from arcticdb_ext.storage import KeyType, Library
+from arcticdb_ext.stream import SegmentInMemory
+from arcticdb_ext.tools import LibraryTool as LibraryToolImpl
+from arcticdb_ext.version_store import AtomKey, PythonOutputFrame, RefKey
+
+
+class LibraryTool:
+    def __init__(self, library: Library):
+        self.impl = LibraryToolImpl(library)
+
+    @staticmethod
+    def key_types() -> List[KeyType]:
+        return list(KeyType.__members__.values())
+
+    @staticmethod
+    def dataframe_to_keys(df: pd.DataFrame, id: Union[str, int], filter_key_type: Optional[KeyType] = None) -> List[AtomKey]:
+        keys = []
+        for index, row in df.iterrows():
+            key_type = KeyType(row["key_type"])
+            if filter_key_type is None or key_type == filter_key_type:
+                keys.append(
+                    AtomKey(
+                        id,
+                        int(row.version_id),
+                        int(row.creation_ts),
+                        int(row.content_hash),
+                        row.start_index.value,
+                        row.end_index.value,
+                        key_type,
+                    )
+                )
+
+        return keys
+
+    def find_keys(self, key_type: KeyType) -> Union[List[AtomKey], List[RefKey]]:
+        return self.impl.find_keys(key_type)
+
+    def find_keys_for_id(self, key_type: KeyType, id: Union[str, int]) -> Union[List[AtomKey], List[RefKey]]:
```

Comment:
Also, if most of the methods are pass-through, I wonder if we could just make this class extend `LibraryToolImpl`? `pybind11` should generate type-annotated Python methods by default.

Reply:
Good idea, will see if it works